pydoit/doit_ex.py

```diff
@@ -0,0 +1,90 @@
+from fn import _
+from helpers import D, F, fifo, to_fifo, group_fastqs
+#NOTE:
+'''
+ they're dropping python2 support in next release
+ it's by modification time
+ the output will always have to be created as a fifo, but not the input
+ to wrap "targets" in fifo
+ 'task_dep' does exist
+'''
+
+class Opts(object): pass
+opts = Opts()
+opts.cutadapt = { 'headcrop' : 33 }
+opts.bwa = { 'threads' : 1, 'keep_temp' : False}
+opts.ref = F("ref.fasta")
+
+'''example tasks'''
+def task_index_ref():
+    return { 'file_dep' : [opts.ref],
+             'actions' : ['bwa index %(dependencies)'],
+             'targets' : [opts.ref.fai] }
```

Comment:
`F` is a class that I made which extends `str` and uses that syntax for adding extensions. It's not pydoit's thing.

F("foo").red == F("foo.red") == "foo.red"